web-server/v0.4/src/routes/Dashboard/RunComparison.js

```diff
@@ -303,6 +331,126 @@ export default class RunComparison extends ReactJS.Component {
     this.setState({ timeseriesDropdownSelected: timeseriesDropdownSelected });
   };
 
+  savePDF = async () => {
+    /* maps through the state array containing the IDs and saves the pdf */
+    var doc = new jsPDF('p', 'mm', [1050, 1475]);
+    var position = 5;
+    var fullIHeight = 0;
+    var promises = this.state.defaultChecked.map(async (check, index) => {
```

Comment:
Instead of using a `map` function here, we can use a `forEach` function which avoids return a new array during each call. For more on the differences between `map` and `forEach`: 

[https://codeburst.io/javascript-map-vs-foreach-f38111822c0f](https://codeburst.io/javascript-map-vs-foreach-f38111822c0f)

Reply:
By using forEach function, I am getting `"undefined is not iterable (cannot read property Symbol(Symbol.iterator))"` error in promise.all